### DISCUSSION TITLE
BLE 5 extended scanning?

### DISCUSSION THREAD

**Discussion by thekurtovic** [h2zero/NimBLE-Arduino#551] [Q&A] [answered]
Hi,
I noticed the BLE 5 example only calls setConnectPhy right before attempting the connection, opposed to setting coded phy at the beginning when starting to scan. Does this library support using coded phy for both scanning and connecting? If so, the example seems a bit misleading IMO

**Comment by h2zero** [comment, marked as answer]
When extended advertising is enabled coded PHY scanning is always performed as well as 1M PHY scanning, no user input is required.

**Comment by thekurtovic** [comment]
Thanks for clearing that up, I missed that option within menuconfig.

**Comment by thekurtovic** [comment]
I'm getting an error after enabling the extended advertising option (tested with extended client example)
`.pio/libdeps/esp32s3/NimBLE-Arduino/src/nimble/nimble/host/src/ble_hs_periodic_sync.c: In function 'ble_hs_periodic_sync_free':
.pio/libdeps/esp32s3/NimBLE-Arduino/src/nimble/nimble/host/src/ble_hs_periodic_sync.c:58:24: error: 'struct ble_npl_event' has no member named 'event'
     if((psync->lost_ev).event != NULL)`

I'm using platformio 6.3.0, arduino-esp32 2.0.9 and esp-idf 4.4.4.
Does BLE extended require idf v5?

**Reply by h2zero** [reply]
If you are using this with IDF you should be using [this repo](https://github.com/h2zero/esp-nimble-cpp) instead.
Can you share your platformio.ini?

**Reply by thekurtovic** [reply]
Oh, sorry. I'm using some combination of arduino and IDF, but now that you mention it, I think it's actually IDF which just has arduino as a component. I'll try that and see how it goes.

These are the relevant parts of the ini file
`[env:esp32s3]
board = esp32-s3-devkitc-1
platform = espressif32 @ ^6.3.0
framework = arduino, espidf

lib_deps = h2zero/NimBLE-Arduino@^1.4.1`

**Reply by thekurtovic** [reply]
All right, after removing the lib_dep, and adding `"C:/Users/User/.platformio/packages/framework-espidf/components/esp-nimble/src"` to the include path & browse path, I'm stuck here.

`Dependency Graph
|-- ESP32 BLE Arduino @ 2.0.0
Building in release mode
Compiling .pio\build\esp32s3\esp-nimble\src\NimBLEDevice.o
C:/Users/User/.platformio/packages/framework-espidf/components/esp-nimble/src/NimBLEDevice.cpp:52:12: fatal error: esp32-hal-bt.h: No such file or directory`

esp32-hal-bt.h is located within "C:/Users/User/.platformio/packages/framework-arduinoespressif32/cores/esp32", which exists within the include+browse paths, so I'm not sure why NimBLEDevice.cpp isn't able to find it.

**Reply by h2zero** [reply]
I'm not sure what's going on there, I have not tried this with platformio (easier to use the idf plugin for vscode btw). I'll try this later and see if I can help.

**Reply by h2zero** [reply]
Tested this and can confirm the build fails, this is caused by platformio not exposing the path to the arduino core files for the idf components to use. The workaround is to modify the the NimBLEDevice.cpp file by commenting out the lines `#  include "esp32-hal-bt.h"` and `btStarted();`, then add the call to `btStarted();` in your main.cpp somewhere (this prevents the arduino core from releasing the BLE memory).

**Reply by thekurtovic** [reply]
Thanks, following those steps resolved it for me.
Also just wanted to say this (NimBLE-Arduino) was working fine the whole time, I think I did something which messed up my environment shortly before I started testing this, compiles fine after cleaning up the Platformio folders. Using the IDF version doesn't show compiler warnings about preprocessor redefinitions so I like this method better.
Thanks for the help. I checked out the IDF plugin too and am thinking about switching over now

**Reply by h2zero** [reply]
I have a patch to then cmakelists for esp-nimble-cpp that allows compiling with platformio to work without modifying the code.
